jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/HttpStream.java

```diff
@@ -221,6 +221,12 @@ public void succeeded()
             getWrapped().succeeded();
         }
 
+        @Override
+        public boolean abort(Throwable cause)
```

Comment:
The HttpStream callbacks and wrappers are only called from the "three legged stool":  after dispatch has returned; after Channel callback has completed; and after any outstanding write has completed.     So it is safe from aborts.   
Moreover, it is already implemented (HTTP1 at least )so that succeeded and failed methods are atomic and multiples ignored.   However, we should review the abort actions of each HttpStream.  Currently HTTP1 closes the endpoint... I guess that will do it.. but....